Number_types/test/Number_types/CMakeLists.txt

```diff
@@ -65,22 +65,26 @@ create_single_source_cgal_program("to_interval_test.cpp")
 create_single_source_cgal_program("unsigned.cpp")
 create_single_source_cgal_program("utilities.cpp")
 find_package(GMP)
-if(GMP_FOUND)
+if( GMP_FOUND AND NOT CGAL_DISABLE_GMP)
   create_single_source_cgal_program("CORE_Expr_ticket_4296.cpp")
   find_package(MPFI)
   if(MPFI_FOUND)
     include(${MPFI_USE_FILE})
   endif(MPFI_FOUND)
 endif(GMP_FOUND)
 # all the programs below will be linked against MPFI in case it is present
-create_single_source_cgal_program("Gmpfi.cpp")
-create_single_source_cgal_program("Gmpfr_bug.cpp")
 create_single_source_cgal_program("Quotient_new.cpp")
 create_single_source_cgal_program("test_nt_Coercion_traits.cpp")
-
-create_single_source_cgal_program("eigen.cpp")
-find_package(Eigen3 3.2.0) #(requires 3.2.0 or greater)
-include(CGAL_Eigen3_support)
-if(TARGET CGAL::Eigen3_support)
-  target_link_libraries(eigen PUBLIC CGAL::Eigen3_support)
+if(CGAL_Core_FOUND)
+  create_single_source_cgal_program( "Gmpfi.cpp" )
+  create_single_source_cgal_program( "Gmpfr_bug.cpp" )
+  create_single_source_cgal_program( "eigen.cpp" )
```

Comment:
It would be clearer is the test was `if(NOT CGAL_DISABLE_GMP)`. That is equivalent, because since CGAL is header-only CGAL_Core is available as soon as GMP is.

Reply:
Assuming it is equivalent, yes, that would be a **lot** clearer...

Reply:
Although even then, it is confusing that with CGAL_DISABLE_GMP we can still have CGAL_USE_MPFI but should ignore it...